pkg/instrumentation/redis/redis.go

```diff
@@ -0,0 +1,191 @@
+package redis
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/redis/go-redis/v9"
+	redistrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/redis/go-redis.v9"
+
+	"github.com/scribd/go-sdk/pkg/cache"
+	"github.com/scribd/go-sdk/pkg/logger"
+)
+
+type (
+	Config struct {
+		ApplicationName string
+		ClientConfig    *cache.Redis
+		Logger          logger.Logger
+	}
+)
+
+const (
+	serviceNameSuffix = "cache-redis"
+)
+
+func New(cfg Config) (redis.UniversalClient, error) {
+	serviceName := fmt.Sprintf("%s-%s", cfg.ApplicationName, serviceNameSuffix)
+
+	opts, err := cfgToRedisClientOptions(cfg.ClientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	c := redis.NewUniversalClient(opts)
```

Comment:
From what I understand, we are creating the Redis client here under the `instrumentation` package.
Why can't we do the same as the database, have a separate `instrumentation` and `cache` packages? We have a [database package](https://github.com/scribd/go-sdk/tree/40f3beb4cb8d108a5cca03e5a06849f495cf3eb9/pkg/database) where we create the connection and then pass it to `instrumentation` package to [instrument](https://github.com/scribd/go-sdk/blob/40f3beb4cb8d108a5cca03e5a06849f495cf3eb9/pkg/instrumentation/database.go#L36) it.

Reply:
@laynax That is a good point. My initial idea was to put everything together, but now I think it makes sense to split the client and instrumentation. Just splitter it 👍